javascript/javascript1/week4/voiceAssistant.js

```diff
@@ -0,0 +1,218 @@
+}
+
+// find out if todoList contains item, if it does return them
+function getTodoItems(){
+    if (todoList != undefined){
+        return "You have " + todoList.length + " todo: " + strTodoItems()
+    }
+    return "todo list is empty"
+}
+
+// get todo items one by one in a str
+function strTodoItems(){
+    let strItems = ''
+    for (let i=0; i<todoList.length; i++){
+        strItems += todoList[i] + ' ' 
+    
+    }
+    return strItems
+}
+
+//function that Handles adding new todo to array todoList 
+function addTodo(){
+    let todoCommand = ''
+    // we loop through the command after the world 'add' til the word 'to'
+    // and add all those words as a todo string and push it to 'todoList' array
+    for(let i=1; i<splitedCommand.length; i++){
+        if (splitedCommand[i] !== "to"){
+            todoCommand += splitedCommand[i] + ' '
+        }
+        else{
+            todoList.push(todoCommand)
+            return todoCommand + " added to your todo"
+        }
+    }
+}
+
+// we loop through the command after the world 'Remove' til the word 'from'
+// and all those words as a todo string and remove it from 'todoList' array if exist
+function removeTodo(splitedCommand){
+    let todoCommand = ''
+    for(let i=1; i<splitedCommand.length; i++){
+        if (splitedCommand[i] !== "from"){
+            console.log(splitedCommand[i])
+            todoCommand += splitedCommand[i] + ''
+        }
+        else{
+            for (let i=0; i<todoList.length; i++){
+                if (todoCommand === todoList[i]){
+                    todoList.splice(todoList[i], todoList[i+1])
+                    return "Removed " + todoCommand + " from your list"
+                }
+            }
+            return todoCommand + " not found in the todo list"
+        }
+    }
+
+}
+
+// functions handles get date command
+function getTodaysDay(){
+    const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+    const d = new Date()
+    const day = d.getDate()
+    const month = months[d.getMonth()]
+    const year = d.getFullYear()
+    return "Today is: " + day + " of " + month + " " + year
+
+}
+
+// function that handles math
+function doMath(){
+    for(let i=0; i<splitedCommand.length; i++){
+        if (splitedCommand[i] === "+"){
+            const result = parseInt(splitedCommand[i-1]) + parseInt(splitedCommand[i+1])
+            return result
+        }
+        else if (splitedCommand[i] === "-"){
+            const result = parseInt(splitedCommand[i-1]) - parseInt(splitedCommand[i+1])
+            return result
+        }
+        else if (splitedCommand[i] === "*"){
+            const result = parseInt(splitedCommand[i-1]) * parseInt(splitedCommand[i+1])
+            return result
+        }
+        else if (splitedCommand[i] === "/"){
+            const result = parseInt(splitedCommand[i-1]) / parseInt(splitedCommand[i+1])
+            return result
```

Comment:
An alternative method to do this would be to convert to array of words like below example
 const commandWords = inputString.split(' ');
  const number1 = parseInt(commandWords[2]);
  const number2 = parseInt(commandWords[4]);
  const operation = commandWords[3];
  if (number1 && number2) {
    switch (operation) {
      case '+':
        return number1 + number2;
      case '-':
        return number1 - number2;
      case '*':
        return number1 * number2;
      case '/':
        return number1 / number2;
    }
  }

Reply:
I see it